### PR TITLE
Add Lucky Orange integration

### LINKED ISSUE #172: Lucky Orange integration
[Lucky Orange](https://www.luckyorange.com) is another website visitor surveillance service. They call themselves, the original all-in-one conversion optimization suite.

Basically, they're similar to Hotjar, but with a more aggressive business model (e.g. only 7 days free tier).

### PR DESCRIPTION
**Pull request by bittner**
Adds a new integration for a service similar to Hotjar.

Fixes #172

### CODE REVIEW

File: analytical/templatetags/luckyorange.py
Comment (PiDelport) [inline analytical/templatetags/luckyorange.py:17]: Is this hostname dynamic, or safe to include statically?

I couldn't find any clear documentation from Lucky Orange about this.

Comment (bittner) [reply analytical/templatetags/luckyorange.py:17]: I don't know either. I though, we'll find out. I would _assume_ this is a generated CloudFront URL that doesn't change. They would have hassle with their clients, otherwise.

I could, of course, open a support ticket with Lucky Orange to clarify this.

Comment (PiDelport) [reply analytical/templatetags/luckyorange.py:17]: Is the code snippet from generic, shared documentation, or somewhere from within an individual account's interface on their service?

If it's from generic, shared documentation, it's probably fine to follow that and update the snippet as the documentation updates: it would be good to link to that from the code.

Otherwise, I'd be worried about the possibility of leaking account context or other account-specific information that might get included unintentionally in this?

It would be good to have official confirmation from Lucky Orange's support that the snippet and hostname is safe and secure to share between different account contexts.

Comment (bittner) [reply analytical/templatetags/luckyorange.py:17]: You're right. You smelled it. It isn't from their documentation but rather from their settings interface in the user account. :+1: 

There are two reasons why I _assume_ that this is neither account-specific nor a secret we may accidentally leak:

1. The code snippet you are meant to include in your site is publicly available (sticking your nose in the generated website HTML code is sufficient).
1. Lucky Orange doesn't feel like a company to me that ia fully into Open Source and transparently publishing details; the lack of the code snippet in the docs might be a result of just that.

BTW, there is a [screenshot with a code snippet](https://help.luckyorange.com/article/123-where-can-i-find-my-tracking-code) in their public documentation, but it's hopelessly outdated. :zany_face: 

I'll to open a support ticket with them and ask for clarification.

Comment (PiDelport) [reply analytical/templatetags/luckyorange.py:17]: Awesome, thank you.

On the balance, I can see that this is _probably_ okay, and that they probably just globally share that hostname (at least for now), but I think that django-analytical should err on the side of caution, and only merge this once we have explicit confirmation that the hostname is safe and secure to share globally, between all users of this project.

Comment (bittner) [reply analytical/templatetags/luckyorange.py:17]: I understand your caution. Though, I would volunteer to sacrifice our Lucky Orange account for getting this merge in early. :smirk: 

If that code wouldn't work _at all_ for other `SITE_ID`s, i.e. other customers, that would be bad, though.

Comment (bittner) [reply analytical/templatetags/luckyorange.py:17]: As it's Sunday and support won't answer immediately, I have taken a look into their [official integrations](https://help.luckyorange.com/article/95-installation-guides), e.g.

| Integration | URL used |
|----------|--------|
| [Drupal module](https://help.luckyorange.com/article/100-how-to-install-in-drupal)<br>(ZIP file in step 4) | Has `... ? "https://ssl" : "http://cdn") + ".luckyorange.com/w.js"` as in the aforementioned screenshot |
| [Joomla! extension](https://www.luckyorange.com/plugins/joomla/installation/installation/index.html)<br>(`joomla.zip` in Step 4) | Has `... ? "https://ssl" : "http://cdn") + ".luckyorange.com/w.js"` as in the aforementioned screenshot |
| [Wordpress plugin](https://wordpress.org/plugins/lucky-orange/)<br>(<kbd>Download</kbd> button) | Has `https://d10lpsik1i8c69.cloudfront.net/w.js` in the source code |

Hard to say for sure, but as Wordpress is the most widely used CMS worldwide it's probably safe to assume that will likely be the most up-to-date integration. It sports the same CloudFront URL we use, so they themselves don't worry.

**Comment by smithdc1** [comment]
I proof read the docs and they seem good to me.

**Comment by bittner** [comment]
Okay, I rebased the branch. So, we only need to decide whether we want to require explicit confirmation from Luky Orange support on the included CDN URL, before merging.

BTW, to implement this integration I simply copy+pasted & search+replaced the excellent Hotjar integration code, including the docs. And for the rebasing I made sure with a `diff` that everything is still super-aligned. So, I'm standing on the shoulders of others who are bigger, smarter and more hardworking than I am. :smile: – Thanks @PiDelport!

File: analytical/templatetags/luckyorange.py
Comment (PiDelport) [reply analytical/templatetags/luckyorange.py:17]: Ah, cool, great catch! Thank you for doing that research. 💜️

I agree: if their official WordPress plugin uses that, we can assume it's safe and secure.

**Review by PiDelport — APPROVED** [review]

**Pull request merged**
